stdlib/ballerina-builtin/src/main/java/org/ballerinalang/nativeimpl/file/service/endpoint/Register.java

```diff
@@ -0,0 +1,148 @@
+import org.ballerinalang.connector.api.Struct;
+import org.ballerinalang.model.types.BType;
+import org.ballerinalang.model.types.TypeKind;
+import org.ballerinalang.nativeimpl.file.service.DirectoryListenerConstants;
+import org.ballerinalang.nativeimpl.file.service.FSListener;
+import org.ballerinalang.natives.annotations.Argument;
+import org.ballerinalang.natives.annotations.BallerinaFunction;
+import org.ballerinalang.natives.annotations.Receiver;
+import org.ballerinalang.util.codegen.PackageInfo;
+import org.ballerinalang.util.codegen.StructInfo;
+import org.wso2.transport.localfilesystem.server.connector.contract.LocalFileSystemConnectorFactory;
+import org.wso2.transport.localfilesystem.server.connector.contract.LocalFileSystemServerConnector;
+import org.wso2.transport.localfilesystem.server.connector.contractimpl.LocalFileSystemConnectorFactoryImpl;
+import org.wso2.transport.localfilesystem.server.exception.LocalFileSystemServerConnectorException;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.ballerinalang.nativeimpl.file.service.DirectoryListenerConstants.FILE_SYSTEM_EVENT;
+import static org.ballerinalang.nativeimpl.file.utils.Constants.FILE_PACKAGE;
+
+/**
+ * Register file listener service.
+ */
+
+@BallerinaFunction(
+        orgName = "ballerina",
+        packageName = "file",
+        functionName = "register",
+        receiver = @Receiver(type = TypeKind.STRUCT, structType = "Listener", structPackage = "ballerina.file"),
+        args = {@Argument(name = "serviceType", type = TypeKind.TYPEDESC)},
+        isPublic = true
+)
+public class Register extends BlockingNativeCallableUnit {
+    @Override
+    public void execute(Context context) {
+        Service service = BLangConnectorSPIUtil.getServiceRegistered(context);
+        Struct serviceEndpoint = BLangConnectorSPIUtil.getConnectorEndpointStruct(context);
+        Struct serviceEndpointConfig = serviceEndpoint
+                .getStructField(DirectoryListenerConstants.SERVICE_ENDPOINT_CONFIG);
+        try {
+            final Map<String, Resource> resourceRegistry = getResourceRegistry(service);
+            final String events = String.join(",", resourceRegistry.keySet());
+            final Map<String, String> paramMap = getParamMap(serviceEndpointConfig, events);
+            LocalFileSystemConnectorFactory connectorFactory = new LocalFileSystemConnectorFactoryImpl();
+            StructInfo structInfo = getStructInfo(context);
+            LocalFileSystemServerConnector serverConnector = connectorFactory
+                    .createServerConnector(service.getName(), paramMap, new FSListener(resourceRegistry, structInfo));
+            serviceEndpoint.addNativeData(DirectoryListenerConstants.FS_SERVER_CONNECTOR, serverConnector);
+        } catch (LocalFileSystemServerConnectorException e) {
+            throw new BallerinaConnectorException("Unable to initialize server connector", e);
```

Comment:
Should we return error here ?

Reply:
@hasithaa what is the correct way to handle this?